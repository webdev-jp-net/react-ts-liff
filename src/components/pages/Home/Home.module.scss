@use 'style/_variable' as *;

.home {
  position: relative;
}

.menu {
  display: grid;
  gap: rem(18) rem(12);
  padding: rem(24);
}
