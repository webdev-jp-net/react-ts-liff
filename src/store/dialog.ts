import { createSlice } from '@reduxjs/toolkit';

type State = {
  isOpen: boolean;
  content: HTMLElement | null;
  class?: string[];
};

const initialState: State = {
  isOpen: false,
  content: null,
};

const dialog = createSlice({
  name: 'dialog',

  initialState,

  reducers: {
    updateOpen: (state, action) => {
      const result: {
        isOpen: boolean;
        class?: string[];
      } = {
        isOpen: action.payload,
      };
      if (action.payload === true) {
        // open
        document.body.classList.add('isFixedScroll');
      } else {
        // close
        document.body.classList.remove('isFixedScroll');
        result.class = [];
      }
      return {
        ...state,
        ...result,
      };
    },
    updateContent: (state, action) => {
      const result: {
        content: HTMLDivElement;
      } = {
        content: action.payload,
      };
      return {
        ...state,
        ...result,
      };
    },
    updateClass: (state, action) => {
      const result: {
        class: string[];
      } = {
        class: [state.class, ...action.payload],
      };
      return {
        ...state,
        ...result,
      };
    },
  },
});

// Action Creators
export const { updateOpen, updateContent, updateClass } = dialog.actions;

// Reducer
export default dialog.reducer;
