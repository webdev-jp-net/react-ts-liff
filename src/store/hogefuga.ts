import { createSlice } from '@reduxjs/toolkit';

type State = {
  dummyText: string;
};

const initialState: State = {
  dummyText: 'ダミーテキスト',
};

const hogefuga = createSlice({
  name: 'hogefuga',

  initialState,

  reducers: {
    updateDummyText: (state, action) => {
      const result: {
        dummyText: string;
      } = {
        dummyText: action.payload,
      };
      return {
        ...state,
        ...result,
      };
    },
  },
});

// Action Creators
export const { updateDummyText } = hogefuga.actions;

// Reducer
export default hogefuga.reducer;
