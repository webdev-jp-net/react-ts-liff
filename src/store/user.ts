import { createSlice } from '@reduxjs/toolkit';

type State = {
  userId?: string;
};

const initialState: State = {};

const user = createSlice({
  name: 'user',

  initialState,

  reducers: {
    updateUserId: (state, action) => {
      state.userId = action.payload;
    },
  },
});

// Action Creators
export const { updateUserId } = user.actions;

// Reducer
export default user.reducer;
