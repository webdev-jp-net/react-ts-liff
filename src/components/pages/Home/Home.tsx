import { FC, useEffect } from 'react';

import { useNavigate } from 'react-router-dom';

import { useSelector } from 'react-redux';

import reportWebVitals from 'reportWebVitals';

import { Button } from 'components/parts/Button';
import { RootState } from 'store';

import { usePageTitle } from 'hooks/usePageTitle';

import styles from './Home.module.scss';

// import { Hoge } from 'components/hoge';

// import { Fuga } from '@types/fuga';

export const Home: FC = () => {
  const navigate = useNavigate();

  const { userId } = useSelector((state: RootState) => state.user);

  // ページを表示したとき
  useEffect(() => {
    reportWebVitals(console.log);
  }, []);

  // ページタイトル
  usePageTitle(`HOME`);

  return (
    <div className={`l-page ${styles.home}`}>
      <h1>HOME</h1>
      <p>ここへページの内容を表示</p>
      <p>{userId}</p>
      <div className={styles.menu}>
        <Button
          handleClick={() => {
            navigate('/sub/1');
          }}
        >
          1のページへ移動
        </Button>
      </div>
    </div>
  );
};
