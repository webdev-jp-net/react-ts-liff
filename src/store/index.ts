import { combineReducers, configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';

import hogefuga from './hogefuga';
import user from './user';

const reducer = combineReducers({
  hogefuga,
  user,
});

const middleware = getDefaultMiddleware({ serializableCheck: false });
export const store = configureStore({ reducer, middleware });

export type RootState = ReturnType<typeof reducer>;
