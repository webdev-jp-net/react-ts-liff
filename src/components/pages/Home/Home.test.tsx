import { Router } from 'react-router-dom';

import { Provider } from 'react-redux';

import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { store } from 'store';

import { Home } from './Home';

const history = createMemoryHistory();
describe('Home', () => {
  beforeEach(() => {
    window.scrollTo = jest.fn();
  });

  it('Homeのスナップショット', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router location={history.location} navigator={history}>
          <Home />
        </Router>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
