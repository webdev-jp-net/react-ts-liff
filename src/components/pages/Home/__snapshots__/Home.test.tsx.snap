// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Homeのスナップショット 1`] = `
<DocumentFragment>
  <div
    class="l-page home"
  >
    <h1>
      HOME
    </h1>
    <p>
      ここへページの内容を表示
    </p>
    <p />
    <div
      class="menu"
    >
      <button
        class="button --basic"
        type="button"
      >
        1のページへ移動
      </button>
    </div>
  </div>
</DocumentFragment>
`;
